lib/src/floodFilling.js

```diff
@@ -0,0 +1,352 @@
+/* eslint-disable */
+import * as THREE from 'three'
+import {OBJLoader} from 'three/examples/jsm/loaders/OBJLoader'
+
+const loader = new OBJLoader();
+
+loader.load(
+  './testdata/testSpace.obj',
+  function ( object ) {
+
+  },
+);
+
+
+
+const box1 = new THREE.Box3(new THREE.Vector3(0, 0, 0), new THREE.Vector3(20, .3, 5))
+const box2 = new THREE.Box3(new THREE.Vector3(0, 0, 0), new THREE.Vector3(.3, 15, 5))
+const box3 = new THREE.Box3(new THREE.Vector3(9.7, 0, 0), new THREE.Vector3(10, 10, 5))
+const box4 = new THREE.Box3(new THREE.Vector3(0, 9.7, 0), new THREE.Vector3(15, 10, 5))
+const box5 = new THREE.Box3(new THREE.Vector3(0, 5, 0), new THREE.Vector3(5, 5.3, 5))
+const box6 = new THREE.Box3(new THREE.Vector3(5, 0, 0), new THREE.Vector3(5, 5.3, 5))
+const box7 = new THREE.Box3(new THREE.Vector3(19.7, 0, 0), new THREE.Vector3(20, 15, 5))
+const box8 = new THREE.Box3(new THREE.Vector3(0, 14.7, 0), new THREE.Vector3(20, 15, 5))
+const box9 = new THREE.Box3(new THREE.Vector3(15, 9.7, 0), new THREE.Vector3(15.3, 15, 5))
+
+
+export let pointsAccepted = []
+export let allPointsAccepted = []
+export let pointsIntersected = []
+export let allPointsIntersected = []
+export let pointsEdge = []
+export let allPointsEdge = []
+export let geoEdges = []
+export let allGeoEdges = []
+let pointsVisited = []
+export let pointsDenied = []
+let pointsLeft = 0
+let FFAttempt = 0
+
+const FFmaxAttempt = 20
+const voxelSize = 1
+const maxThreshold = 10000
+
+
+export const boundingBoxesAll = [box1, box2, box3, box4, box5, box6, box7, box8, box9]
+
+RecursiveFloodFilling()
+
+export async function FloodFilling() {
+  const IFCUnionBox = calculateIFCBoundingBox(boundingBoxesAll)
+  SatisfyFirstRandom(pointsAccepted, pointsVisited, voxelSize, IFCUnionBox, boundingBoxesAll)
+
+  pointsLeft ++
+
+  let intVis = 0
+  while ((pointsLeft>0)&&(intVis<maxThreshold)) {
+    try {
+      CheckVoxelSurrounding(pointsAccepted[intVis], voxelSize, IFCUnionBox, boundingBoxesAll)
+    } catch {
+      console.log('Error was found on pAccepted '+intVis)
```

Comment:
catch and log should be at least console.warn or perhaps console.error

Reply:
This would be in the FloodFilling PR